### PR TITLE
ci: use duckdb instead of ibis to test interchange-only support

### PR DESCRIPTION
**Pull request by MarcoGorelli**
as discussed in https://github.com/vega/altair/pull/3591#issuecomment-2453480628

### CODE REVIEW

File: tests/vegalite/v5/test_api.py [hunk 1]
```diff
@@ -1607,51 +1607,72 @@ def test_polars_with_pandas_nor_pyarrow(monkeypatch: pytest.MonkeyPatch):
...
 
 
-@pytest.mark.skipif(
-    Version("1.5") > PANDAS_VERSION,
-    reason="A warning is thrown on old pandas versions",
-)
-@pytest.mark.xfail(
-    sys.platform == "win32", reason="Timezone database is not installed on Windows"
-)
-def test_ibis_with_vegafusion(monkeypatch: pytest.MonkeyPatch):
```
Comment (dangotbanned) [inline tests/vegalite/v5/test_api.py:1638]: @MarcoGorelli I'm assuming dropping this was unintentional?

In (https://github.com/vega/altair/pull/3672/commits/c32e1eb625919a46e4b1a3b04e257cb279b83330) I've moved some things around to make this harder to miss in the future.

File: tests/vegalite/v5/test_api.py [hunk 2]
```diff
@@ -1607,51 +1607,72 @@ def test_polars_with_pandas_nor_pyarrow(monkeypatch: pytest.MonkeyPatch):
...
-    tbl = ibis.memtable(df)
+
+    # Narwhals doesn't fully support our custom DuckDBWithInterchangeSupport,
+    # so we need to overwrite `to_native`
+    def to_native(df, strict):
+        if isinstance(df, nw.DataFrame):
+            return rel_df
+        return df
+
+    monkeypatch.setattr("narwhals.stable.v1.to_native", to_native)
```
Comment (dangotbanned) [inline tests/vegalite/v5/test_api.py:1670]: I'm really quite impressed by this 👍

**Review by dangotbanned — APPROVED** [review]
Thank you so much @MarcoGorelli!

Pretty blown away by how quickly you put all this together 🤯

File: tests/vegalite/v5/test_api.py (hunk 1 above)
Comment (MarcoGorelli) [reply tests/vegalite/v5/test_api.py:1638]: As in, the skipif? I was just curious to see if it was an ibis-specific thing and was still needed

**Pull request merged**